src/components/Plans/UpdatePaymentInformation/UpdatePaymentInformation/Reprocess/index.js

```diff
@@ -0,0 +1,135 @@
+              </a>
+            </th>
+            <th>
+              <a href="#">
+                {_('updatePaymentMethod.reprocess.invoices_declined_table.invoice_column')}
+              </a>
+            </th>
+            <th>
+              <a href="#">
+                {_('updatePaymentMethod.reprocess.invoices_declined_table.amount_column')}
+              </a>
+            </th>
+          </tr>
+        </thead>
+        <tbody>
+          {declinedInvoices.map((invoice) => (
+            <tr role="row" aria-label="invoice" key={invoice.invoiceNumber}>
+              <td>
+                <FormattedDate value={invoice.date} />
+              </td>
+              <td>{invoice.invoiceNumber}</td>
+              <td>
+                {dollarSymbol} <FormattedNumber value={invoice.amount} {...numberFormatOptions} />
+              </td>
+            </tr>
+          ))}
+        </tbody>
+      </table>
+    </>
+  );
+};
+
+export const Reprocess = InjectAppServices(({ dependencies: { dopplerBillingUserApiClient } }) => {
+  const intl = useIntl();
+  const [state, setState] = useState({ loading: true, declinedInvoices: { invoices: [] } });
+  const [status, setStatus] = useState('');
+  const [messageError, setMessageError] = useState('');
+  const createTimeout = useTimeout();
+  const _ = (id, values) => intl.formatMessage({ id: id }, values);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const declinedInvoices = await dopplerBillingUserApiClient.getDeclinedInvoices();
+
+      setState({
+        declinedInvoices: declinedInvoices.success ? declinedInvoices.value : { invoices: [] },
+        loading: false,
+      });
+    };
+    fetchData();
+  }, [dopplerBillingUserApiClient]);
+
+  const reprocess = async () => {
+    const result = await dopplerBillingUserApiClient.reprocess();
+
+    if (result.success) {
+      setStatus(SAVED);
+
+      createTimeout(() => {
+        window.location.href = `login`;
+      }, DELAY_BEFORE_REDIRECT_TO_SUMMARY);
+    } else {
+      setMessageError(handleMessage(result.error));
+      setStatus(HAS_ERROR);
+    }
+  };
+
+  const showMessage = [SAVED, HAS_ERROR].includes(status);
+
+  return (
+    <>
+      {state.loading ? (
+        <Loading page />
+      ) : (
+        <>
+          <div role="pending-amount-section" aria-label="pending-ammount">
+            {_('updatePaymentMethod.reprocess.pending_amount_message')} {': '} {dollarSymbol}{' '}
+            <FormattedNumber value={state.declinedInvoices.totalPending} {...numberFormatOptions} />
+          </div>
+          <DeclinedInvoices declinedInvoices={state.declinedInvoices.invoices} />
+          {showMessage && (
+            <StatusMessage
+              type={status === SAVED ? 'success' : 'cancel'}
+              message={_(
+                status === SAVED ? 'updatePaymentMethod.reprocess.success_message' : messageError,
+              )}
```

Comment:
We have not a translation for the returned `messageError`

Reply:
Yes, we have the translation inside ```first_data_error```

Reply:
oh, grate!

in that case, what do you think about renaming  `messageError` as `errorMessageKey`?

Reply:
Yes, I agree with you. I will change by errorMessageId